### PR TITLE
pagination design

### LINKED ISSUE #1200: Use buttons for pagination
Replace current pagination on the search page with a button set.

The current pagination does not have a left and right arrow, but it would be nice to add them as per below.

```
<div class="button-set">
  <a class="wh-button disabled">Previous</a>
  <a href="" class="wh-button -highlight">1</a>
  <a href="" class="wh-button">2</a>
  <a href="" class="wh-button">3</a>
  <a href="" class="wh-button disabled">...</a>
  <a href="" class="wh-button">500</a>
  <a href="" class="wh-button">Next</a>
</div>
```
We'll need to translate 'previous' and 'next'.

Also, 'previous/next buttons should only be disabled if you are on the first/last page :)

### PR DESCRIPTION
**Pull request by nkanaev**
resolves #1200 

result:
screen shot 2016-06-28 at 14 29 42

### CODE REVIEW

**Comment by dstufft** [comment]
It looks like the test had failed due to some network hickups or something. I want ahead and restarted them and they're all green now. This looks good from my end, however @nlhkabu is going to have to review this since it's front end/design work.

**Comment by nlhkabu** [comment]
Hi @nkanaev, thanks for this PR.

My preference here is to use a button group, so that we don't have to add specific CSS for the pagination (less code, less to maintain). It's also more visually consistent with the rest of the design :)

Based on the work you have done, is there any reason why the button group markup wouldn't work?

**Comment by nkanaev** [comment]
i do agree that less css is better. i'd actually implemented it as a button set at first. but then noticed making "..." elements as unclickable links doesn't make sense, so I needed a logical way to add `span` element. in the end i've ended up implementing something bem'ish (though it's not Gallagher's one).

do have any suggestions on spans? maybe we should just use something like:

    <span class="wh-button">...</span>

**Comment by nlhkabu** [comment]
Yes, let's try using a span with a button class.  If necessary, we can modify the button BEM, but it *should* work out of the box :)

You can also apply a disabled modifier to the span, e.g.

```
<span class="button button--disabled">...</span>
```

This will make it grey and give it a `not-allowed` cursor.

File: warehouse/static/sass/blocks/_button-set.scss [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    margin-left: -2px;
+    border-radius: 0;
+
+    &:first-child {
+      border-radius: 3px 0 0 3px;
+    }
+
+    &:last-child {
+      border-radius: 0 3px 3px 0;
+      z-index: 5;
```
Comment (nkanaev) [inline warehouse/static/sass/blocks/_button-set.scss]: hacky, had to create to resolve the overlapping border (grey color) of the last element while on the last page.

**Comment by nkanaev** [comment]
final result:
screen shot 2016-07-04 at 14 43 29

**Comment by nlhkabu** [comment]
Thanks for upading this @nkanaev :)

I'm still not convinced that a new button **set** is needed.  I'd much rather modify the existing button **group** to see if we can make it work.

Does this markup do the trick?

```
<div class="button-group"
  <a class="button button-group__button">Previous</a>
  <a class="button button-group__button">1</a>
  <a class="button button-group__button button--highlight">2</a>
  <a class="button button-group__button">3</a>
  <a class="button button-group__button">4</a>
  <span class="button button-group__button button--disabled">...</a>
  <a class="button button-group__button">382</a>
  <a class="button button-group__button">Next</a>
</div>
```

**Comment by nkanaev** [comment]
thanks, totally have missed `button-group`.
had to adjust a little bit - other buttons were overlapping highlight buttons on mouse hover, which is visible only when overlapping button is set to disabled.

**Comment by nlhkabu** [comment]
Merged :sparkles: 

Thanks very much for this @nkanaev :)

**Pull request merged**